### PR TITLE
perf: use CompositionTarget to only render a specific target

### LINKED ISSUE #14345: [Skia] `CompositionTarget`-based rendering
### What would you like to be added

Currently we have a single `Compositor` and each visual change will invalidate all rendering surfaces (even across multiple `XamlRoots`). However, we should instead invalidate and render just a single `CompositionTarget` which matches the current visual tree.

### Why is this needed

Rendering performance.

### For which platform

Skia (WPF), Skia (GTK on Linux/macOS/Windows), Skia (Linux Framebuffer)

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by ramezgerges**
GitHub Issue (If applicable): closes #14345

## PR Type

What kind of change does this PR introduce?

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Internal/ContentRoot.cs [hunk 1]
```diff
@@ -68,6 +69,10 @@ public ContentRoot(ContentRootType type, Color backgroundColor, UIElement? rootE
 			FocusAdapter = new FocusAdapter(this);
 			FocusManager.SetFocusObserver(new FocusObserver(this));
 
+			CompositionTarget = new CompositionTarget();
+			CompositionTarget.Root = VisualTree.RootElement.Visual;
```
Comment (MartinZikmund) [inline src/Uno.UI/UI/Xaml/Internal/ContentRoot.cs:73]: To make this more concise, I think the `Root` setter could automatically set the `CompositionTarget` as well, thus the following line's logic would happen in the setter

**Review by MartinZikmund — APPROVED** [review]

**Comment by jeromelaban** [comment]
@MartinZikmund I missed that it was targeting your branch. Merge as needed, though it will need to be built on your target branch since the CI does not cover such merge target validations,

File: src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Composition/CompositionTarget.cs [hunk 2]
```diff
@@ -8,25 +8,6 @@ namespace Windows.UI.Composition
 #endif
 	public partial class CompositionTarget : global::Windows.UI.Composition.CompositionObject
 	{
-#if __ANDROID__ || __IOS__ || IS_UNIT_TESTS || __WASM__ || __SKIA__ || __NETSTD_REFERENCE__ || __MACOS__
```
Comment (MartinZikmund) [inline src/Uno.UWP/Generated/3.0.0.0/Windows.UI.Composition/CompositionTarget.cs:11]: Undo this change and adjust the one in `Microsoft.UI.Xaml.Media` instead. Then the implementation will need to be inside Uno.UI project, so the `Visual` in `Uno.UI.Composition` will just use plain `object` to reference the instance (as composition project does not have reference to `Uno.UI`)
